commit de2d76f264b8ff3528a4946bee94ba63473c08d6
Author: Daniel Axtens <dja@axtens.net>
Date:   Fri Mar 3 17:56:55 2017 +1100

    crypto: powerpc - Fix initialisation of crc32c context
    
    commit aa2be9b3d6d2d699e9ca7cbfc00867c80e5da213 upstream.
    
    Turning on crypto self-tests on a POWER8 shows:
    
        alg: hash: Test 1 failed for crc32c-vpmsum
        00000000: ff ff ff ff
    
    Comparing the code with the Intel CRC32c implementation on which
    ours is based shows that we are doing an init with 0, not ~0
    as CRC32c requires.
    
    This probably wasn't caught because btrfs does its own weird
    open-coded initialisation.
    
    Initialise our internal context to ~0 on init.
    
    This makes the self-tests pass, and btrfs continues to work.
    
    Fixes: 6dd7a82cc54e ("crypto: powerpc - Add POWER8 optimised crc32c")
    Cc: Anton Blanchard <anton@samba.org>
    Signed-off-by: Daniel Axtens <dja@axtens.net>
    Acked-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1460_de2d76f264b8ff3528a4946bee94ba63473c08d6.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1460_de2d76f264b8ff3528a4946bee94ba63473c08d6.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
